src/main/java/webserver/HttpRequest.java

```diff
@@ -0,0 +1,76 @@
+package webserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import utils.HttpRequestBody;
+import utils.HttpRequestHeader;
+import utils.HttpRequestMethod;
+import utils.HttpRequestPath;
+import utils.HttpRequestQueryString;
+import utils.IOUtils;
+import utils.RequestParserUtils;
+
+public class HttpRequest {
+    private final HttpRequestMethod httpRequestMethod;
+    private final HttpRequestPath httpRequestPath;
+    private final HttpRequestQueryString httpRequestQueryString;
+    private final HttpRequestHeader httpRequestHeader;
+    private final HttpRequestBody httpRequestBody;
+
+    private HttpRequest(HttpRequestMethod httpRequestMethod, HttpRequestPath httpRequestPath, HttpRequestQueryString httpRequestQueryString, HttpRequestHeader httpRequestHeader, HttpRequestBody httpRequestBody) {
+        this.httpRequestMethod = httpRequestMethod;
+        this.httpRequestPath = httpRequestPath;
+        this.httpRequestQueryString = httpRequestQueryString;
+        this.httpRequestHeader = httpRequestHeader;
+        this.httpRequestBody = httpRequestBody;
+    }
+
+    public HttpRequestMethod getHttpRequestMethod() {
+        return httpRequestMethod;
+    }
+
+    public HttpRequestPath getHttpRequestPath() {
+        return httpRequestPath;
+    }
+
+    public HttpRequestQueryString getHttpRequestQueryString() {
+        return httpRequestQueryString;
+    }
+
+    public HttpRequestHeader getHttpRequestHeader() {
+        return httpRequestHeader;
+    }
+
+    public HttpRequestBody getHttpRequestBody() {
+        return httpRequestBody;
+    }
+
+    public static HttpRequest ofFirstLine(BufferedReader bufferedReader) throws IOException {
+        String line = bufferedReader.readLine();
+
+        if (line == null) { return null;}
```

Comment:
습관적으로 IDE의 코드 자동정렬을 수행해주세요. 🙏 (mac: `⌥ + ⌘ + L`,  window: `Ctrl + Alt + L`)

```suggestion
        if (line == null) {
            return null;
        }
```